code/game/objects/items/devices/scanners.dm

```diff
@@ -253,9 +258,15 @@ GENE SCANNER
 
 	if(ishuman(M))
 		var/mob/living/carbon/human/H = M
-		var/ldamage = H.return_liver_damage()
-		if(ldamage > 10)
-			to_chat(user, "\t<span class='alert'>[ldamage > 45 ? "Severe" : "Minor"] liver damage detected.</span>")
+		var/damage
+		var/high_threshold
+		var/low_threshold
+		for(var/obj/item/organ/organ in H.internal_organs)

```

Comment:
if there can only be /.../organ types here make this typeless

Reply:
Fairly certain using a typeless var won't work here, but I'll check.

Reply:
```DM
(var/thingy in stuff)
var/the/only/type/in/stuff/varname = thingy
```

Reply:
Why, though? The objects stored are already children of the organ object, and if for whatever reason they're not we don't want them. Pulling the data out as typeless to typecast it right afterwards seems a tad unnecessary, unless this'll fix the issue with "the [organ]" instead of "[organ]", which I forgot to mention isn't really a problem now that there's unique organ failure messages.

Reply:
see contrib.md, it's faster.